core/src/main/scala/org/apache/spark/sql/crossdata/XDSession.scala

```diff
@@ -45,14 +45,10 @@ class XDSession(
   // TODO move context to sharedState => userConfig should be read before. Otherwise properties will be the same for all
   // TODO we can't move XDContext to XDSession because RunnableCommand use XDContext instead of SparkSession.sessionCatalog
 
-  // xdContext will host common Crossdata context entities
-  //private val xdContext: XDContext = XDContext.getOrCreate(sc, userConfig) //Delegated XDContext
-
-
   //TODO: +1 Use catalog for this session instead fix one
   override protected[sql] lazy val catalog: XDCatalog = {
-      val catalogs: List[XDCatalogCommon] =  xdSessionState.xdTemporaryCatalog :: externalCatalog :: streamingCatalog.toList
-      CatalogChain(catalogs:_*)(catalystConf)
+    val catalogs: Seq[XDCatalogCommon] = (xdSessionState.temporaryCatalogs :+ externalCatalog) ++ streamingCatalog.toSeq
```

Comment:
Why changing the building underlying collection when using a list would be more concise?
`val catalogs: Seq[XDCatalogCommon] =  xdSessionState.xdTemporaryCatalog :: externalCatalog :: streamingCatalog.toList`